src/services/APIClient.ts

```diff
@@ -16,40 +17,91 @@ export class APIClient {
     });
   }
 
-  async getBlocks(options?: { offset?: number, limit?: number }): Promise<IBlock[]> {
+  async getBlocks(options?: {
+    offset?: number;
+    limit?: number;
+  }): Promise<IBlock[]> {
     const response = await this.axios.get<IBlock[]>('/blocks', {
-      params: options
+      params: options,
     });
 
     return response.data;
   }
 
   async getBlock(blockId: string): Promise<IBlock> {
-    const response = await this.axios.get<{data: IBlock}>(`/blocks/${blockId}`);
+    const response = await this.axios.get<{ data: IBlock }>(
+      `/blocks/${blockId}`
+    );
 
     return response.data.data;
   }
 
   async getLeavesOfBlock(blockId: string): Promise<ILeaf[]> {
-    const response = await this.axios.get<ILeaf[]>(`/blocks/${blockId}/leaves`); 
+    const response = await this.axios.get<ILeaf[]>(`/blocks/${blockId}/leaves`);
 
     return response.data;
   }
 
   async getKeys(): Promise<IKey[]> {
-    const response = await this.axios.get<{data: IKey[]}>('/keys');
+    const response = await this.axios.get<{ data: IKey[] }>('/keys');
     return response.data.data;
   }
 
   async getProofs(keys: string[]): Promise<IProofs | null> {
-    const response = await this.axios.get<{data: IProofs | Record<string, never>}>('/proofs', {
-      params: { keys }
+    const response = await this.axios.get<{
+      data: IProofs | Record<string, never>;
+    }>('/proofs', {
+      params: { keys },
     });
 
-    if(response.data.data.block) {
+    if (response.data.data.block) {
       return response.data.data as IProofs;
     }
 
     return null;
   }
+
+  /**
+   * Uses verifyProofForBlock method of the Chain contract.
+   * @see https://kovan.etherscan.io/address/0x459DF121Ab6Cf1B4C99119cc354d7C79c83Ec8bE#readContract
+   */
+  async verifyProofForBlock<T extends LeafType>(
+    key: string,
+    leafType: T
+  ): Promise<{
+    success: boolean;
+    value: T extends typeof LeafType.TYPE_INTEGER
+      ? number
+      : T extends typeof LeafType.TYPE_FLOAT
+      ? number
+      : string;
+  }> {
+    if (!this.options.chainContract) {
+      throw new Error('chainContract is required');
+    }
+
+    const proofs = await this.getProofs([key]);
+
+    if (!proofs) {
+      throw new Error('No block found');
+    }
+
+    const verified = await this.options.chainContract.verifyProofForBlock({
+      blockHeight: proofs.block.height,
+      proofs: proofs.leaves[0].proof,
+      key,
+      value: proofs.leaves[0].value,
+      leafType,
+    });
+
+    return {
+      success: verified,
+      value: (leafType === LeafType.TYPE_INTEGER
+        ? parseInt(proofs.leaves[0].value, 10)
+        : leafType === LeafType.TYPE_FLOAT
+          ? parseFloat(proofs.leaves[0].value)
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          : proofs.leaves[0].value) as any,
```

Comment:
instead of disabling the rule, don't we know the type(s) for that value? function has a type, we should be able to use the same type, no?

Reply:
@DZariusz i thought that typescript could realize that we return the right types, but it were showing an error, looks like typescript can't correctly resolve conditional types. I'll try to fix the situation with function overloading